t/t5516-fetch-push.sh

```diff
@@ -1712,4 +1712,23 @@ test_expect_success 'updateInstead with push-to-checkout hook' '
 	)
 '
 
+test_expect_success 'denyCurrentBranch and worktrees' '
+    git worktree add new-wt &&
+	git clone . cloned &&
+	test_commit -C cloned first &&
+	test_config receive.denyCurrentBranch refuse &&
+	test_must_fail git -C cloned push origin HEAD:new-wt &&
+	test_config receive.denyCurrentBranch updateInstead &&
+	git -C cloned push origin HEAD:new-wt &&
+	test_must_fail git -C cloned push --delete origin new-wt
+'
+
+test_expect_failure 'denyCurrentBranch and unborn branch' '
+	git init repo &&
```

Comment:
Maybe call it "unborn" instead of "repo"? The former would give a much stronger hint.